### ISSUE TITLE
Does not properly convert nested lists

### ISSUE THREAD

**Issue by btusakul** [Phil-Factor/PSYaml#24]
Given a file with this content: 

```
- test1
-   - test2
    - test3
```

the expected output from piping this file's content to ConvertFrom-Yaml would be a list with 'test1' and a nested list with 'test2' and 'test3'. This should be the proper formatting for nested lists in YAML and converting this YAML file in Python, with the PyYAML module, exhibits this behavior. 

However, the ConvertFrom-Yaml cmdlet returns a list with 'test1', 'test2', and 'test3'.

**Comment by Phil-Factor** [comment]
Yes, this is valid yaml and should be represented as you describe. it is missing the second '-' on that line. Unfortunately this is happening in aaubry's YamlDotNet library. Curiously, the parser appears to get it right but at some point the result is converted to a string array with three components.
